app/services/subscriptions/dates/monthly_service.rb

```diff
@@ -12,7 +12,7 @@ def compute_from_date(date = base_date)
       end
 
       def compute_charges_from_date
-        if terminated?
+        if subscription.terminated_at?(billing_at)
```

Comment:
Here it was using the method defined as:
```ruby
def terminated?
  subscription.terminated? && !subscription.next_subscription
end
```

Maybe we should keep the `&& !subscription.next_subscription` as well (meaning upgrade this `terminated?` itself) what do you think?

Reply:
ohh good spot 👍 - thanks!